cartographer/mapping/internal/2d/pose_graph_2d.cc

```diff
@@ -554,7 +559,7 @@ void PoseGraph2D::RunOptimization() {
   // is time consuming, so not taking the mutex before Solve to avoid blocking
   // foreground processing.
   optimization_problem_.Solve(constraints_, frozen_trajectories_,
-                              landmark_nodes_);
+                              landmark_nodes_, freeze_landmarks_);
```

Comment:
Update comment above, add freeze_landmarks to the list of variables that we promise no other thread is accessing.